File: src/TypeScript/TypeScriptLibrary/Library/Measurements/RecursiveFormula.ts

import { IDesktop } from "../Interfaces/IDesktop";
import { IMeasurements } from "./Interfaces/IMeasurements";
import { IPostSetArrow } from "../Interfaces/IPostSetArrow";
import { DataConsumerVariadbleMeasurementsStarted } from "./DataConsumerVariadbleMeasurementsStarted";


export class RecursiveFormula extends DataConsumerVariadbleMeasurementsStarted implements IPostSetArrow
{
    protected inputs: IMeasurements[] = [];


    protected arguments: string[] = [];

  //  protected initial: Map<string, any> = new Map();

    protected operationNames: Map<number, string> = new Map();


    constructor(desktop: IDesktop, name: string)
    {
        super(desktop, name);
        this.typeName = "RecursiveFormula";
        this.types.push("IPostSetArrow");
        this.types.push("RecursiveFormula");

    }

 
    init(): void
    {

    }

   
    postSetArrow(): void
    {
        this.init();
        this.setInitial();
        this.setFeedback();
    }

    getAllMeasurements(): IMeasurements[] {
        return this.inputs;
    }

    addMeasurements(item: IMeasurements): void {
        this.inputs.push(item);
    }

    calculateTree(): void
    {

    }

    save(): void
    {

    }


    updateMeasurements(): void
    {
        this.feedback.setFeedBackAliases();
        this.calculateTree();
        this.save();
    }
}

File: Java_Generated/Android/measurements/RecursiveFormula.java
package measurements;

import diagram.interfaces.IDesktop;
import general_service.interfaces.IFeedbackCollection;
import measurements.interfaces.IFeedbackHolder;
import measurements.interfaces.IMeasurements;
import measurements.interfaces.IStarted;
import measurements.variables.Variable;


import java.util.HashMap;
import java.util.Map;

public class RecursiveFormula extends DataConsumerVariableMeasurementsStarted implements IStarted, IMeasurements, IFeedbackHolder
{
    public RecursiveFormula(String name, IDesktop desktop) {
        super(name, desktop);
    }

    @Override
    public void postSetArrow() {
        init();
        createInitial();
        createFeedback();
    }

    @Override
    public IMeasurements[] getAllMeasurements() {
        return inputs;
    }

    @Override
    public void addMeasurements(IMeasurements item)
    {
        inputs = performer.extend(inputs, item);
    }


    @Override
    public void startedStart(double time)
    {
        initial.resetInitialValues();
        feedback.setFeedbacks();
    }

    protected void calculateTree()
    {

    }


    protected void save()
    {

    }

    @Override
    public void  updateMeasurements()
    {
        calculateTree();
        save();
        feedback.setFeedbacks();
    }


    protected  IMeasurements[]  inputs = new IMeasurements[0];

    @Override
    public IFeedbackCollection getFeedbackCollection() {
        return feedback;
    }
}
